### PR TITLE
Removing preloader

### PR DESCRIPTION
**Pull request by umbobabo**
Change the logic adding anyway the fallback to remove the preloader.
In this way, also we can't load the external script, the fallback is available.

### CODE REVIEW

File: index.es6 [hunk 1]
```diff
@@ -119,26 +127,10 @@ export default class GoogleSearch extends React.Component {
...
-              <input type="hidden" name="cx"
-                value="${this.props.cx}" id="edit-cx"
-              />
-            </form>`;
         this.setState({
-          fallbackHTML,
+          useFallback: true,
         });
+        this.focusSearchField();
         console.error('An error occurs loading or executing Google Custom Search: ', e.message);
```
Comment (fabiosantoscode) [inline index.es6]: You need to rethrow the error here, otherwise this turns into a resolved promise.

**Comment by fabiosantoscode** [comment]
Serverside rendering should also set `useFallback:true`. You should also check if `typeof window !== 'undefined'` before using promisescript. It's a lot easier to have a function that returns the promise to the script, something like

```
function googleScript() {
  if (typeof window == 'undefined') { return Promise.reject('cannot load scripts in node'); }
  return promisescript({ url: theUrl }).then(() => { assert(window.whateverWeExpectToLoad, 'script didnt actually load'); return window.whateverWeExpectToLoad; })
}
```

This function can be called however many times you want because promisescript keeps a cache. If you want to cache it from your side, you can `import once from 'once'`.

File: index.es6 [hunk 2]
```diff
@@ -97,21 +104,22 @@ export default class GoogleSearch extends React.Component {
...
-
     if (!this.script) {
       window.__gcse = {
         parsetags: 'explicit',
-        callback: myCallback,
+        callback: () => {
+          if (document.readyState === 'complete') {
+            renderSearchElement();
+          } else {
+            google.setOnLoadCallback(renderSearchElement, true);
```
Comment (fabiosantoscode) [inline index.es6]: Is this `if` necessary? I can't find many docs for setOnLoadCallback, and the ones I *can* find don't say if you need to check first.

Comment (fabiosantoscode) [inline index.es6]: It's not important, though old IE doesn't like "readyState" and it's a huge bummer.

**Comment by umbobabo** [comment]
I set on the SS to use the fallback by default, we can't trust to show them only if loading the script fails.
What happens with promescript anyway is on demand on user interaction so I don't think we have to check in which side we are.

File: index.es6 [hunk 3]
```diff
@@ -44,7 +43,8 @@ export default class GoogleSearch extends React.Component {
     this.state = {
       statusClassName: 'search--close',
       searchTerm: '',
-      fallbackHTML: '',
+      // useFallback by default on SS
+      useFallback: (typeof window === 'undefined'),
```
Comment (fabiosantoscode) [inline index.es6]: I love it

File: index.es6 (hunk 1 above)
Comment (fabiosantoscode) [inline index.es6]: ok super cool

**Comment by umbobabo** [comment]
@fabiosantoscode the href is a Brrrrrrillliant solution!
Unfortunately the external script dynamically create is own input field so we don't have control on this.
QuerySelector is now quite supported http://caniuse.com/#feat=queryselector

File: index.es6 [hunk 4]
```diff
@@ -69,77 +71,62 @@ export default class GoogleSearch extends React.Component {
       searchTerm: '',
       statusClassName: 'search--close',
     });
-    if (typeof document.querySelector('.search input.gsc-input') !== 'undefined') {
-      document.querySelector('.search input.gsc-input').value = '';
+    if (this.state.useFallback) {
```
Comment (fabiosantoscode) [inline index.es6]: You need to check for `googleSearchInputFallbackInput` here. There's a delay between setting the state and rendering, and if componentShouldUpdate is broken, render won't be called again.

Comment (fabiosantoscode) [inline index.es6]: nvm

**Comment by fabiosantoscode** [comment]
:rainbow:

**Pull request merged**